test/uri/uri_canonicalize_test.cc

```diff
@@ -68,10 +68,16 @@ TEST(URI_canonicalize, example_10) {
   EXPECT_EQ(uri.recompose(), "urn:example:schema");
 }
 
-TEST(URI_canonicalize, example_11) {
+TEST(URI_canonicalize, example_relative_1) {
   sourcemeta::jsontoolkit::URI uri{"../foo"};
   uri.canonicalize();
-  EXPECT_EQ(uri.recompose(), "/foo");
+  EXPECT_EQ(uri.recompose(), "foo");
```

Comment:
Regarding what I see [here](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4): 

```
If the input buffer begins with a prefix of "../" or "./",
then remove that prefix from the input buffer; otherwise,

I think we should keep it like that.
```

Reply:
@tony-go You are reading a section under reference resolution. But on `canonicalize` I think that's fine too :+1: